cli/inputhandler/jsonfile.cpp

```diff
@@ -0,0 +1,43 @@
+#include "jsonfile.h"
+#include "utils/errors.h"
+
+#include <iostream>
+#include <fstream>
+
+namespace ih
+{
+JsonFile::JsonFile(std::string const &filePath) :
+        IFile(filePath)
+{
+    try
+    {
+        JsonFile::checkFile();
+    }
+    catch (std::exception const &error)
+    {
+        throw;
+    }
+}
+
+bool JsonFile::checkFile() const
+{
+    if (!IFile::fileExists()) throw std::invalid_argument(ERROR_MSG_FILE_DOES_NOT_EXIST);
+    if (!IFile::isFileExtension("json")) throw std::invalid_argument(ERROR_MSG_FILE_EXTENSION_INVALID);
+
+    return true;
```

Comment:
Perhaps we can drop the `bool` from the signature - and perhaps even drop the function `checkFile()` from the `IFile` interface, and only keep it in implementations?

Reply:
Removed the bool return and made it void.
Not sure about it, since the whole purpose of making `checkFile` pure virtual is to "force" the classes that derive from IFile to implement this method. This way, when creating a new object that derives from IFile, you  are forced in the constructor to check for file validity and throw errors if needed. We can ofc change this when we implement in/out file factories